pageserver/src/bin/pageserver.rs

```diff
@@ -320,18 +322,33 @@ fn start_pageserver(
     // Scan the local 'tenants/' directory and start loading the tenants
     BACKGROUND_RUNTIME.block_on(mgr::init_tenant_mgr(conf, remote_storage.clone()))?;
 
+    // shared state between the background task and the http endpoint; note that the http endpoint
+    // is still accessible even if background task is not configured as long as remote storage has
+    // been configured.
+    let disk_usage_eviction_state: Arc<DiskUsageEvictionState> = Arc::default();
+
     if let Some(remote_storage) = &remote_storage {
-        launch_disk_usage_global_eviction_task(conf, remote_storage.clone())?;
+        launch_disk_usage_global_eviction_task(
+            conf,
+            remote_storage.clone(),
+            disk_usage_eviction_state.clone(),
+        )?;
     }
```

Comment:
Would like to see this as:

```patch
    let disk_usage_eviction_state = if let Some(remote_storage) = &remote_storage {
        let state: Arc<DiskUsageEvictionState> = Arc::default();
        launch_disk_usage_global_eviction_task(
            conf,
            remote_storage.clone(),
            disk_usage_eviction_state.clone(),
        )?;
        Some(state)
    }
```

Then later on, add this as an `Option` instead of `Arc`, allowing the use of builder methods at least some place.. Or create the http router state in this file. Right now `http::routes::State::new` exists, which hurts and is unnecessary making these changes much more verbose.

Reply:
Perhaps I should do this pretty much next, because it'll be forgotten otherwise (making the state be built in main).

Reply:
Created issue about this #3894.